main.py

```diff
@@ -24,31 +49,52 @@ def init_background(self, background_symbol, width, height):
       
 current_frame = Frame()
 
-def draw(x,y,symbol=None,symbols=None):
+def draw(x,y,symbol=None,symbols=None,use_screen_space=False):
+    global world_x
+    global world_y
     if symbols == None:
-        i = int((height-y) * (width) + (x-1))
+        screen_x = x 
+        screen_y = y
+        if use_screen_space == False:
+            screen_x-=world_x
+            screen_y-=world_y
+        if screen_x <1 or screen_y <1 or screen_x > width or screen_y > height:
+            return
+        i = int((height-screen_y) * (width) + (screen_x-1))
         current_frame.image = current_frame.image[:i] + symbol + current_frame.image[i + 1:]
     else:
         for symbol in symbols:
-            i = int((height-y-symbol[1]+1) * (width) + (x+symbol[0]-2))
+            screen_x = x+symbol[0]-1
+            screen_y = y+symbol[1]-1
+            if use_screen_space == False:
+                screen_x-=world_x
+                screen_y-=world_y
+            if screen_x <1 or screen_y <1 or screen_x > width or screen_y > height:
+                continue
+            i = int((height-screen_y) * (width) + (screen_x-1))
             current_frame.image = current_frame.image[:i] + symbols[symbol] + current_frame.image[i + 1:]
 
 def render_frame():
     print(current_frame.banner+current_frame.image)
 
 def game():
+    global world_x
+    global world_y
     character_hash_map = {tuple([2,1]):'|', tuple([8,1]):'|', tuple([3,2]):'_',tuple([4,2]):'_',tuple([5,2]):'|',tuple([6,2]):'_',tuple([7,2]):'_', tuple([1,3]):'|',tuple([5,3]):'|',tuple([9,3]):'|',tuple([2,4]):'_', tuple([3,4]): '_',tuple([4,4]):'_',tuple([5,4]):'|',tuple([6,4]):'_',tuple([7,4]):'_',tuple([8,4]):'_', tuple([5,5]):'>'                        
 ,tuple([4,6]):'0',tuple([6,6]):'0',tuple([4,7]):'~',tuple([6,7]):'~'}
-
+    object_hash_map = {tuple([1,1]):'_', tuple([2,1]):'_', tuple([3,1]):'_',tuple([1,2]):'|',tuple([3,2]):'|',tuple([1,3]):'_',tuple([2,3]):'_', tuple([3,3]):'_'}
     for num in range(10000):
-        current_frame.init_background('*',width, height)        
+        current_frame.init_background('*',width, height) 
+        get_input()       
         draw(width/2,height/2,symbol='c')
         draw(1,1,symbol='L')
         draw(width,height,symbol='R')
         draw(width,1,symbol='d')
         draw(1,height,symbol='u')
-        draw(width/2,height/2,symbols=character_hash_map)
+        draw(width/2,height/2,symbols=character_hash_map,use_screen_space=True)
+        draw(1,height/2,symbols=object_hash_map)
         render_frame()
+
```

Comment:
add a function called 

> update_frame_dimensions
that uses the terminal height and width and is called in every for loop

Reply:
added function